### PR TITLE
Refresh plugin for August 2023 (upgrade minimum core version to 2.361.4 and Java version to 11)

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

**Comment by basil** [comment]
Currently blocking https://github.com/jenkinsci/bom/pull/2440

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,4 @@
-buildPlugin(
-  useContainerAgent: true,
-  configurations: [
-    [platform: 'linux', jdk: 8],
-    [platform: 'windows', jdk: 8],
+buildPlugin(useContainerAgent: true, configurations: [
+    [platform: 'linux', jdk: 17],
+    [platform: 'windows', jdk: 11],
```
Comment (olamy) [inline Jenkinsfile:3]: still java 8 support?
if not maybe worth to add the minimum jdk requirement in the PR title as is this will be in the release notes

**Review by olamy — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (basil) [reply Jenkinsfile:3]: Am I understanding correctly that you are a maintainer of this repository and can therefore edit PR titles?

**Comment by basil** [comment]
Currently blocking https://github.com/jenkinsci/bom/pull/2440

File: Jenkinsfile (hunk 1 above)
Comment (olamy) [reply Jenkinsfile:3]: I''m really sorry for the delay and misunderstanding. But as you are the owner of the current work (this PR) I was under the impression you would have been the most appropriate to modify your own work after review from the member of this project's community. I'm not usually modifying work from others.
But so as it looks to take time, I will do it.

**Comment by olamy** [comment]
I modified the title. I really hope it doesn't modify too much your work with your agreement.

**Comment by basil** [comment]
Thanks. I generally don't include minimum core version changes in release notes for the plugins I maintain (unless the minimum core version is unusual, like a weekly release or a very recent LTS) because such plugin releases aren't even advertised in the UI to users of older cores. Similarly, a recent core version implies a minimum Java version, and users running older Java versions (and therefore older cores) wouldn't get an advertisement for such a plugin upgrade in the UI, so I generally don't include minimum Java version changes in release notes. I have filed dozens (hundreds?) of PRs without this information in the title and shipped dozens of releases without this information in the release notes, all without any complaints from users or other developers. But I acknowledge that as the plugin maintainer, you should be free to include (or not include) any information you want in the release notes. I hope that this discussion about writing release notes has not been a significant distraction from the comparatively far more laborious task of modernizing the plugin itself.

**Pull request merged**